### ISSUE TITLE
Document or explain "<uninhabited>"

### ISSUE THREAD

**Issue by ddfisher** [python/mypy#3030]
Users can now get `<uninhabited>` in error messages when there are no constraints on a type variable.  This should either be documented somewhere, or we should use a more self-explanatory string to display the type.

**Comment by ethanhs** [comment]
I would say a better more explanatory error is a better option. I didn't know what it meant till reading more about it, and I expect many others would be similarly confused.

**Comment by ddfisher** [comment]
Agreed.  Though actually the better option still might be to not infer UninhabitedType when there are no constraints on a type variable.  (See #3032.)

**Comment by ilevkivskyi** [comment]
I just renamed the issue since `UninhabitedType()` is currently formatted as `<nothing>`.

**Comment by GabrielMajeri** [comment]
I'm trying to determine where would be the proper place to document this type. 
Would the [Kind of types](https://github.com/python/mypy/blob/master/docs/source/kinds_of_types.rst) page be the correct location for it?

**Comment by JukkaL** [comment]
> Would the Kind of types page be the correct location for it?

These are related to type variables, so somewhere here would be a better location:

https://github.com/python/mypy/blob/master/docs/source/generics.rst

**Comment by patrick-nicodemus** [comment]
Has there been any progress on this in the past five years?
I have `Revealed type is "<nothing>"` as an error message and I am not sure how to interpret it.
